src/sparseml/optim/helpers.py

```diff
@@ -195,6 +195,68 @@ def parse_recipe_variables(
     return recipe_variables
 
 
+def _update_staged_container_value(var_key, var_value, container):
+    """
+    Extends the functionality of _update_container_value to handle staged containers.
+
+    :param var_key: The key we are attempting to find in the container.
+    :param var_value: The value which will overwrite the previous value
+        every time var_key is found in the container's keys.
+    :param container: A container generated from a YAML string of SparseML recipe
+    :return: (optionally mutated) container, as well as key_value
+        (True if var_key found in container's attributes, otherwise False)
+    """
+    key_found = False
+
+    for container_key, container_value in container.items():
+        if not isinstance(container_value, dict):
+            # checking contents of global variables
+            if var_key == container_key:
+                container[var_key] = var_value
+                key_found = True
+
+        else:
+            # checking contents of a stage
+            stage_container, stage_key_found = _update_container_value(
+                var_key, var_value, container_value
+            )
+            container[container_key] = stage_container
+            if stage_key_found:
+                key_found = True
+
+    return container, key_found
+
+
+def _update_container_value(var_key: str, var_value, container):
+    """
+    Checks whether there is at least one attribute (key) in the container, with the same
+    name as var_key. If this is the case, key_found = True, otherwise False.
+    Additionally, during checking, when var_key key is present in any of the container's
+    attributes, this attribute's value gets overwritten with var_value.
+
+    :param var_key: The key we are attempting to find in the container.
+    :param var_value: The value which will overwrite the previous value
+        every time var_key is found in the container's keys.
+    :param container: A container generated from a YAML string of SparseML recipe
+    :return: (optionally mutated) container, as well as key_value
+        (True if var_key found in container's attributes, otherwise False)
+    """
```

Comment:
nit: pydocs only should be added for public methods, for private, can give inline comments about logic

Reply:
I really enjoy those situations where I get to learn more about the best rules for creating docstrings/pydocs. 

From my naive perspective, it makes sense to document some private methods if it potentially adds value. Afaik, as long as there is an underscore preceding the class/function name, this block of code will not be recognised by pydoc, so no harm here (https://stackoverflow.com/questions/12040381/using-pydoc-for-non-public-private-classes). Additionally, for some private classes/funcs it seems beneficial to add some more verbose documentation, especially for new joining ml-engs in the future.

What do you think?

Reply:
agreed that verbose documentation even for private methods is important. Keeping them out of the triple quote pydoc format will prevent them from being found and added to our official docs when we build docs eventually.

Reply:
Understood.